etna/models/tbats.py

```diff
@@ -83,6 +86,124 @@ def get_model(self) -> Model:
+
+        seasonal_periods = self._model.seasonal_periods
+        if (
+            seasonal_periods is not None
+            and len(seasonal_periods) > 0
+            and len(fitted_model_params.seasonal_periods) == 0
+        ):
+            warn("Seasonal components is not fitted!")
+
+        if self._model.use_arma_errors and not fitted_model_params.use_arma_errors:
+            warn("ARMA components is not fitted!")
+
+        if self._model.use_box_cox and not fitted_model_params.use_box_cox:
+            warn("Box-Cox transform is not fitted!")
+
+        if self._model.use_trend and not fitted_model_params.use_trend:
+            warn("Trend is not fitted!")
+
+    def _decompose_forecast(self, horizon: int) -> np.ndarray:
+        """Estimate raw forecast components."""
+        if self._fitted_model is None or self._freq is None:
+            raise ValueError("Model is not fitted! Fit the model before estimating forecast components!")
+
+        model = self._fitted_model
+        state_matrix = model.matrix.make_F_matrix()
+        component_weights = model.matrix.make_w_vector()
+
+        state = model.x_last
+        components = []
+        for _ in range(horizon):
+            components.append(component_weights * state)
+            state = state_matrix @ state
+
+        components = np.stack(components, axis=0)
+
+        if model.params.components.use_box_cox:
+            transformed_pred = np.sum(components, axis=1)
+            pred = model._inv_boxcox(transformed_pred)
+            components = components * pred[..., np.newaxis] / transformed_pred[..., np.newaxis]
+
+        return components
+
+    def _named_components(self, raw_components: np.ndarray) -> Dict[str, np.ndarray]:
+        """Prepare components with names."""
+        params_components = self._fitted_model.params.components  # type: ignore
+        named_components = dict()
+
+        named_components["local_level"] = raw_components[:, 0]
+
+        component_idx = 1
+        if params_components.use_trend:
+            named_components["trend"] = raw_components[:, component_idx]
+            component_idx += 1
+
+        if len(params_components.seasonal_periods) != 0:
+            seasonal_periods = params_components.seasonal_periods
+
+            if hasattr(params_components, "seasonal_harmonics"):
+                seasonal_harmonics = params_components.seasonal_harmonics
+                for seasonal_period, seasonal_harmonic in zip(seasonal_periods, seasonal_harmonics):
+                    named_components[f"seasonal(s={seasonal_period})"] = np.sum(
+                        raw_components[:, component_idx : component_idx + 2 * seasonal_harmonic], axis=1
+                    )
+                    component_idx += 2 * seasonal_harmonic
+
+            else:
+                component_idx -= 1
+                for seasonal_period in seasonal_periods:
+                    component_idx += seasonal_period
```

Comment:
In my code i did sth like this `component_idx += seasonal_period - 1`, this is not the same as `component_idx -= 1`

Reply:
Yes, in your code you had wrong offsets for cases with more than one  seasonal periods for TBATS